### PR TITLE
TF uses ros::MessageEvent to get connection information

### PR DESCRIPTION
**Pull request by kwatts**
TF should stop using the "__connection_header" field of ROS messages, but instead use the ros::MessageEvent.

### CODE REVIEW

**Comment by dirk-thomas** [comment]
This is mandatory for a _Indigo_ release. Backporting it to _Hydro_ (and older) is optional.

**Comment by wjwwood** [comment]
Is there any justification for the reason to change this, is there some other ticket I am un aware of?

@dirk-thomas why is this mandatory for Indigo? Is it just cleaner or is there some use of deprecated functionality that we expect to be gone in Indigo?

**Comment by kwatts** [comment]
The "__connection_header" field for each message is going away in Indigo:
https://github.com/ros/gencpp/commit/2c3d57bdb93bda2c5828d80b848601e57b8e7f08

After that, only the ros::MessageEvent can be used to get connection info.

**Comment by wjwwood** [comment]
Was this deprecated in Hydro?

**Comment by wjwwood** [comment]
This is more relevant than the commit: https://github.com/ros/gencpp/issues/3

**Comment by tfoote** [comment]
Merged into indigo-devel

File: tf/src/tf_monitor.cpp [hunk 1]
```diff
@@ -63,23 +63,10 @@ class TFMonitor
   tf::tfMessage message_;
 
   boost::mutex map_lock_;
-  void callback(const tf::tfMessageConstPtr& msg_ptr)
+  void callback(const ros::MessageEvent<tf::tfMessage>& msg_evt)
```
Comment (dirk-thomas) [inline tf/src/tf_monitor.cpp:66]: Shouldn't this be:

```
void callback(const ros::MessageEvent<tf::tfMessage const>& msg_evt)
```

Comment (tfoote) [inline tf/src/tf_monitor.cpp:66]: Fixed in 7d9fee121557d06e533509a7537924ad84ea4158

**Pull request closed without merging**